dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/task/dependent/DependentLogicTask.java

```diff
@@ -72,34 +70,24 @@ public DependentLogicTask(TaskExecutionContext taskExecutionContext,
 
     @Override
     public AsyncTaskExecuteFunction getAsyncTaskExecuteFunction() {
-        return new DependentAsyncTaskExecuteFunction(taskExecutionContext,
+        dependentAsyncTaskExecuteFunction = new DependentAsyncTaskExecuteFunction(taskExecutionContext,
                 taskParameters,
                 projectDao,
                 workflowDefinitionDao,
                 taskDefinitionDao,
                 taskInstanceDao,
                 workflowInstanceDao);
+        return dependentAsyncTaskExecuteFunction;
     }
 
     @Override
     public void pause() throws MasterTaskExecuteException {
-        if (workflowExecutionRunnable == null) {
-            log.error("Cannot find the WorkflowExecuteRunnable");
-            return;
-        }
-        TaskInstance taskInstance = workflowExecutionRunnable
-                .getWorkflowExecuteContext()
-                .getWorkflowExecutionGraph()
-                .getTaskExecutionRunnableById(taskExecutionContext.getTaskInstanceId())
-                .getTaskInstance();
-        if (taskInstance == null) {
-            log.error("Cannot find the TaskInstance in workflowExecuteRunnable");
-            return;
-        }
-        taskInstance.setState(TaskExecutionStatus.PAUSE);
-        taskInstance.setEndTime(new Date());
-        taskInstanceDao.upsertTaskInstance(taskInstance);
-        super.pause();
+        // todo: support pause
```

Comment:
Dependent Task is very important in ds, so I think this is a high priority todo.

If it cannot be completed in a short time, I suggest to add some warn logs here.

Reply:
Yes, will use https://github.com/apache/dolphinscheduler/issues/16577 to fix this.